src/Pact/Native/Db.hs

```diff
@@ -377,16 +390,26 @@ write wt partial i as = do
 
 createTable' :: GasRNativeFun e
 createTable' g i [t@TTable {..}] = do
-  guardTable i t
+  guardTable i t GtCreateTable
   let (UserTables tn) = userTable t
   computeGas' g i (GPreWrite (WriteTable (asString tn))) $
     success "TableCreated" $ createUserTable (_faInfo i) tn _tModuleName
 createTable' _ i as = argsError i as
 
-guardTable :: Pretty n => FunApp -> Term n -> Eval e ()
-guardTable i TTable {..} = guardForModuleCall (_faInfo i) _tModuleName $
-  enforceBlessedHashes i _tModuleName _tHash
-guardTable i t = evalError' i $ "Internal error: guardTable called with non-table term: " <> pretty t
+guardTable :: Pretty n => FunApp -> Term n -> GuardTableOp -> Eval e ()
+guardTable i TTable {..} dbop =
+  checkLocalBypass $
+    guardForModuleCall (_faInfo i) _tModuleName $
+      enforceBlessedHashes i _tModuleName _tHash
+  where checkLocalBypass notBypassed = do
+          localBypassEnabled <- isExecutionFlagSet FlagAllowReadInLocal
+          case dbop of
+            GtWrite -> notBypassed
+            GtCreateTable -> notBypassed
+            _ | localBypassEnabled -> return ()
+              | otherwise -> notBypassed
```

Comment:
I figured explicit was better in the general case, in case we decide against some operation in the future, or allow slice-and-dice in config in the future

Reply:
Alright, that sounds good